Update: update uboot menu script

Pratiksha0102 opened the pull request:
Add kernel Image-specific DTB file handling in extlinux.conf

Pratiksha0102 committed:
u-boot-update: Use version specific kernel dtb

- Updated the script to correctly match DTB files with their
respective kernel Image versions. Previously, the FDT device
tree was linked to rk-kernel.dtb, which caused issues when
multiple kernels were present. Now, the correct DTB file is
selected based on the kernel version.

Signed-off-by: Pratiksha0102 <pratikshathanki2501@gmail.com>

djkabutar reviewed (changes requested):
change commit message to something specific, what was the problem, what have you change and what this change will fix, mention this things!

Pratiksha0102 committed:
u-boot-update: Use kernel specific overlays dir

- Modify the script to correctly add FDT overlays
to extlinux.conf based on the kernel Image version.

- Previously, there was only one directory for
overlays. Now, the overlays directory will vary
according to kernel versions.

Signed-off-by: Pratiksha0102 <pratikshathanki2501@gmail.com>

djkabutar commented on u-boot-update:
```diff
@@ -179,19 +190,29 @@ for _KERNEL in ${_KERNELS}; do
+
+	for dtb in ${U_BOOT_FDT}; do
+		# Check if the DTB matches the current kernel version
+		_IMG_VERSION=$(echo "${_KERNEL#*-}" | sed 's/-.*//') 
```
make this _IMG_VERSION=$(echo "${_KERNEL#*-}" | sed 's/-.*[a-z]+//')

djkabutar commented on u-boot-update:
```diff
@@ -179,19 +190,29 @@ for _KERNEL in ${_KERNELS}; do
+	for dtb in ${U_BOOT_FDT}; do
+		# Check if the DTB matches the current kernel version
+		_IMG_VERSION=$(echo "${_KERNEL#*-}" | sed 's/-.*//') 
+		DTB_VERSION=$(echo "$dtb" | sed -n 's/.*-\([0-9]\+\.[0-9]\+\.[0-9]\+\).*/\1/p')
```
make this DTB_VERSION=$(echo "$dtb" | sed -n 's/.*-\([0-9]\+\.[0-9]\+\.[0-9]\+-[a-z]\+\).*/\1/p')

djkabutar reviewed (changes requested):
do not use words like update, fixed in commit header message. whatever you have done write that only, other changes I have mentioned do that as well.

djkabutar reviewed (changes requested):
do rebase to the first commit and do the last mentioned changes, make sure to change commit message of previous commits as well

Pratiksha0102 committed:
debian: bump debian version to 5.0.3

Signed-off-by: Pratiksha0102 <pratikshathanki2501@gmail.com>

djkabutar reviewed (approved):
versioning scheme for multiple kernels
